src/Mastodon/Http.elm

```diff
@@ -275,50 +288,59 @@ searchAccounts client query limit resolve =
 
 userAccount : Client -> Request Account
 userAccount client =
-    authRequest client GET ApiUrl.userAccount accountDecoder
+    authRequest client GET (client.server ++ ApiUrl.userAccount) accountDecoder
 
 
 postStatus : Client -> StatusRequestBody -> Request Status
 postStatus client statusRequestBody =
-    authRequest client POST ApiUrl.statuses statusDecoder
+    authRequest client POST (client.server ++ ApiUrl.statuses) statusDecoder
         |> Build.withJsonBody (statusRequestBodyEncoder statusRequestBody)
 
 
 deleteStatus : Client -> Int -> Request Int
 deleteStatus client id =
-    authRequest client DELETE (ApiUrl.status id) <| Decode.succeed id
+    authRequest client DELETE (client.server ++ (ApiUrl.status id)) <| Decode.succeed id
 
 
 context : Client -> Int -> Request Context
 context client id =
-    authRequest client GET (ApiUrl.context id) contextDecoder
+    authRequest client GET (client.server ++ (ApiUrl.context id)) contextDecoder
 
 
 reblog : Client -> Int -> Request Status
 reblog client id =
-    authRequest client POST (ApiUrl.reblog id) statusDecoder
+    authRequest client POST (client.server ++ (ApiUrl.reblog id)) statusDecoder
 
 
 unreblog : Client -> Int -> Request Status
 unreblog client id =
-    authRequest client POST (ApiUrl.unreblog id) statusDecoder
+    authRequest client POST (client.server ++ (ApiUrl.unreblog id)) statusDecoder
 
 
 favourite : Client -> Int -> Request Status
 favourite client id =
-    authRequest client POST (ApiUrl.favourite id) statusDecoder
+    authRequest client POST (client.server ++ (ApiUrl.favourite id)) statusDecoder
 
 
 unfavourite : Client -> Int -> Request Status
 unfavourite client id =
-    authRequest client POST (ApiUrl.unfavourite id) statusDecoder
+    authRequest client POST (client.server ++ (ApiUrl.unfavourite id)) statusDecoder
 
 
 follow : Client -> Int -> Request Relationship
 follow client id =
-    authRequest client POST (ApiUrl.follow id) relationshipDecoder
+    authRequest client POST (client.server ++ (ApiUrl.follow id)) relationshipDecoder
 
 
 unfollow : Client -> Int -> Request Relationship
 unfollow client id =
-    authRequest client POST (ApiUrl.unfollow id) relationshipDecoder
+    authRequest client POST (client.server ++ (ApiUrl.unfollow id)) relationshipDecoder
+
+
+
+-- NEW STUFF
+
+
+fetchStatusList : Client -> Method -> String -> Request (List Status)
+fetchStatusList client method url =
+    authRequest client method url <| Decode.list statusDecoder
```

Comment:
I'm planning using this kind of response-type-oriented functions everywhere, and eventually remove `fetchUserTimeline` and friends. Pro: It's more generic and composable. Con: it may be more verbose for the end user to build the url prior to passing it to these functions.

I think we'd really benefit from such a generalized approach though.